nncf/quantization/fake_quantize.py

```diff
@@ -175,28 +175,61 @@ def asymmetric_range(min_values: np.ndarray, max_values: np.ndarray,
 
 
+def get_quantizer_narrow_range(quantizer_config: QuantizerConfig,
+                               quant_group: QuantizerGroup, half_range: bool = False) -> bool:
+    """
+    Returns narrow_range parameter: True if the range of quantized values is reduced by 1 compared to the
+        naive case, False otherwise.
+
+    :param quantizer_config: Config of the quantization configuration.
+    :param quant_group: Group of the quantizer.
+    :param half_range: If True effectively only a half of a quantizer range is used.
+        False - the full range is used.
+    :return: narrow_range parameter.
+    """
+    if quantizer_config.mode == QuantizationMode.SYMMETRIC:
+        return quant_group == QuantizerGroup.WEIGHTS and quantizer_config.num_bits == 8 and not half_range
+    return False
+
+
 def calculate_quantizer_parameters(statistics: MinMaxTensorStatistic,
                                    quantizer_config: QuantizerConfig,
-                                   quant_group: QuantizerGroup) -> FakeQuantizeParameters:
+                                   quant_group: QuantizerGroup,
+                                   narrow_range: bool,
+                                   half_range: bool = False) -> FakeQuantizeParameters:
     """
     Calculates FakeQuantize layer attributes for weight/activation quantizer.
 
     :param statistics: Collected statistics for the quantized insertion.
     :param quantizer_config: Config of the quantization configuration.
     :param quantizer_group: Group of the quantizer.
+    :param narrow_range: True if the range of quantized values is reduced by 1 compared to the
+        naive case, False otherwise.
+    :param half_range: If True effectively only a half of a quantizer range is used.
+        False - the full range is used.
     :return: Parameters of the FakeQuantize layer.
     """
     min_values = np.array(statistics.min_values).astype(np.float32)
     max_values = np.array(statistics.max_values).astype(np.float32)
 
+    num_bits = quantizer_config.num_bits - 1 if half_range else quantizer_config.num_bits
     if quantizer_config.mode == QuantizationMode.SYMMETRIC:
-        narrow_range = quant_group == QuantizerGroup.WEIGHTS
-        _, _, levels = calculate_symmetric_level_ranges(quantizer_config.num_bits,
-                                                        signed=True, narrow_range=narrow_range)
-        input_low, input_high = symmetric_range(min_values, max_values,
-                                                levels, quantizer_config, quant_group)
+        _, _, levels = calculate_symmetric_level_ranges(num_bits, signed=True, narrow_range=narrow_range)
+        input_low, input_high = symmetric_range(min_values, max_values, levels, quantizer_config, quant_group)
+        if half_range:
+            if quant_group == QuantizerGroup.WEIGHTS:
+                _, _, export_levels = calculate_symmetric_level_ranges(
+                    num_bits + 1, signed=True, narrow_range=True
```

Comment:
What is happened if narrow_range=False was passed in the function?

Reply:
Then the model will be exported in 256 levels for weight quantizers which leads to a significant accuracy drop.

Reply:
Refactor the function